File: mobile_flutter/lib/app/format/currency_format.dart
/// Presentation-only formatting matching app.js's two currency formatters
/// exactly (Milestone 6 Web-reference spec, app.js:2923/3311) — pure display
/// string construction, no rounding/business decision beyond the same
/// `Math.round` already used for on-screen display in the Web app.
///
/// [formatHomeCurrency]: unsigned "₪12,345" / "-₪500" — used for plain
/// balances/totals (hero, snapshot, tiles).
/// [formatSignedCurrency]: always-signed "+₪12,345" / "-₪500" — used only
/// for directional cash-flow events, so a positive amount is never
/// ambiguous with a plain balance.
library;

String formatHomeCurrency(num n) {
  final rounded = n.round();
  final grouped = _groupThousands(rounded.abs());
  return rounded < 0 ? '-₪$grouped' : '₪$grouped';
}

String formatSignedCurrency(num n) {
  final rounded = n.round();
  final grouped = _groupThousands(rounded.abs());
  return rounded < 0 ? '-₪$grouped' : '+₪$grouped';
}

String _groupThousands(int n) {
  final digits = n.toString();
  final buffer = StringBuffer();
  for (var i = 0; i < digits.length; i++) {
    if (i > 0 && (digits.length - i) % 3 == 0) buffer.write(',');
    buffer.write(digits[i]);
  }
  return buffer.toString();
}

File: mobile_expo/src/core/currencyFormat.ts
// Presentation-only currency strings, matching the Web app's two display
// formatters (the authoritative reference; also ported by the Flutter oracle
// in lib/app/format/currency_format.dart):
//   unsigned: "₪12,345" / "-₪500"   (balances, totals)
//   signed:   "+₪12,345" / "-₪500"  (directional cash-flow events)
//
// Rounding uses JavaScript Math.round, exactly like app.js. (Dart's
// num.round() rounds .5 away from zero, so the oracle differs from the Web for
// negative half values such as -2.5; the Web is authoritative.)
//
// The returned string is wrapped in an LTR isolate so it renders correctly
// inside Hebrew text; use stripIsolates() to compare raw text.

import { ltr } from './bidi.ts';

function groupThousands(n: number): string {
  const digits = String(n);
  let out = '';
  for (let i = 0; i < digits.length; i++) {
    if (i > 0 && (digits.length - i) % 3 === 0) out += ',';
    out += digits[i];
  }
  return out;
}

function roundedParts(n: number): { negative: boolean; grouped: string } {
  const rounded = Math.round(n);
  return { negative: rounded < 0, grouped: groupThousands(Math.abs(rounded)) };
}

export function formatAmount(n: number): string {
  const { negative, grouped } = roundedParts(n);
  return ltr(negative ? `-₪${grouped}` : `₪${grouped}`);
}

export function formatSignedAmount(n: number): string {
  const { negative, grouped } = roundedParts(n);
  return ltr(negative ? `-₪${grouped}` : `+₪${grouped}`);
}
